src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/BodyModelBinder.cs

```diff
@@ -80,14 +81,15 @@ public BodyModelBinder(IList<IInputFormatter> formatters, IHttpRequestStreamRead
                 throw new ArgumentNullException(nameof(readerFactory));
             }
 
-            _formatters = formatters;
-            _readerFactory = readerFactory.CreateReader;
-
-            if (loggerFactory != null)
+            if (loggerFactory == null)
             {
-                _logger = loggerFactory.CreateLogger<BodyModelBinder>();
+                throw new ArgumentNullException(nameof(loggerFactory));
```

Comment:
I agree this makes the `BodyModelBinder` consistent with the rest. But, this is a change in behaviour that isn't forced: No longer allows users to pass `null` for this parameter.

Unless @Eilon is fine with the unnecessary break, revert changes to this class except the new logging calls and use `?.` in the new logging calls.

Reply:
How is this a breaking change? This would have blown up on this line anyway if `loggerFactory` is null, no?

Reply:
(Yeah the exception type changed, but I'm not concerned about someone "expecting" a `NullRefException` - those are always bugs.

Reply:
The deleted code meant an `ILogger` was created only if `loggerFactory` were non-`null`. And, `_logger?....` avoided any NREs. Bottom line, this `ArgumentNullException` is brand new.

Reply:
Oh sorry I see what you're saying. GitHub wasn't showing enough of the diff in the discussion tab. This is indeed a breaking change and I don't think a good one, so I recommend preserving the old behavior (no exceptions thrown; just ignore null logger).